silverback/application.py

```diff
@@ -132,19 +182,28 @@ def add_taskiq_task(handler: Callable) -> AsyncTaskiqDecoratedTask:
             #       we only want to determine if it is not None
             if container is not None and isinstance(container, ContractEvent):
                 # Address is almost a certainty if the container is being used as a filter here.
-                if contract_address := getattr(container.contract, "address", None):
-                    labels["contract_address"] = contract_address
-                labels["event_signature"] = f"{container.abi.signature}"
+                if not (contract_address := getattr(container.contract, "address", None)):
+                    raise InvalidContainerTypeError(
+                        "Please provider a contract event from a valid contract instance."
+                    )
+
+                labels["contract_address"] = contract_address
+                labels["event_name"] = container.abi.name
+                event_abi = container.abi
```

Comment:
I started with the signature (and could revert back to it), is there a method now to get `EventABI` from a signature?

Reply:
ah, I guess `TaskData` is a message playload now so :shrug:   But yes, those methods were in use in the code you replaced.

Reply:
I actually don't see `EventABI.from_signature` anywhere in the old code, but it is [generally available](https://github.com/ApeWorX/ethpm-types/blob/main/ethpm_types/abi.py#L335C9-L335C23) now so I will use it because I agree it's actually a more compact representation anyways and it fits in labels instead of adding a whole new union type to `TaskData`

Reply:
I double checked, I think you meant to do that but never integrated it once ethpm-types was released (checked it on main)

Reply:
[f392e27](https://github.com/ApeWorX/silverback/pull/82/commits/f392e27ba675163331f1670ebe28a333ccee9f4f)

Reply:
Ah, that went into the broker code in the cluster, not here.  Thought I saw it in the diff, guess it was a hallucination.